front/lib/api/assistant/conversation/validate_actions.ts

```diff
@@ -182,24 +184,29 @@ export async function validateAction(
     return new Ok(undefined);
   }
 
-  await launchAgentLoopWorkflow({
-    auth,
-    agentLoopArgs: {
-      agentMessageId,
-      agentMessageVersion,
-      conversationId,
-      conversationTitle,
-      userMessageId,
-      userMessageVersion,
-    },
-    // Resume from the step where the action was created.
-    startStep: agentStepContent.step,
-    // Wait for completion of the agent loop workflow that triggered the
-    // validation. This avoids race conditions where validation re-triggers the
-    // agent loop before it completes, and thus throws a workflow already
-    // started error.
-    waitForCompletion: true,
-  });
+  if (shouldRunAgentLoop) {
+    await launchAgentLoopWorkflow({
+      auth,
+      agentLoopArgs: {
+        agentMessageId,
+        agentMessageVersion,
+        conversationId,
+        conversationTitle,
+        userMessageId,
+        userMessageVersion,
+      },
+      // Resume from the step where the action was created.
+      startStep: agentStepContent.step,
+      // Wait for completion of the agent loop workflow that triggered the
+      // validation. This avoids race conditions where validation re-triggers the
+      // agent loop before it completes, and thus throws a workflow already
+      // started error.
+      waitForCompletion: true,
+    });
+  } else {
+    // if we don't run the agent loop, we need to clear the action required status manually.
+    await ConversationResource.clearActionRequired(auth, conversationId);
```

Comment:
@flvndvd I was testing this and if we don't launch agent loop, the agent message status will stay as `created`, I think we will end up in a weird state if we manually update the status no? 🤔